ironfish/src/network/peers/peerManager.ts

```diff
@@ -165,6 +188,21 @@ export class PeerManager {
 
     const peer = this.getOrCreatePeer(null)
     peer.setWebSocketAddress(url.hostname, url.port)
+
+    const address = peer.getWebSocketAddress()
+    const peerCandidate = this.peerCandidateMap.get(address)
+    if (!peerCandidate) {
+      this.peerCandidateMap.set(address, {
+        address: url.hostname,
+        port: url.port,
+        neighbors: new Set(),
+        webRtcRetry: new ConnectionRetry(),
+        websocketRetry: new ConnectionRetry(),
+        localRequestedDisconnectUntil: null,
+        peerRequestedDisconnectUntil: null,
+      })
+    }
+
     peer.isWhitelisted = isWhitelisted
```

Comment:
I think we could probably move `isWhitelisted` onto `ConnectionRetry`. That's the only place that we use it is when we call `failedConnection`

Reply:
We decided to move `isWhitelisted` onto `ConnectionRetry`